virttest/env_process.py

```diff
@@ -1525,13 +1526,13 @@ def _take_screendumps(test, params, env):
                 os.makedirs(screendump_dir)
             except OSError:
                 pass
-            counter[vm.instance] += 1
-            filename = "%04d.jpg" % counter[vm.instance]
+            counter[instance] += 1
+            filename = "%04d.jpg" % counter[instance]
```

Comment:
if `vm.instance` changes, `counter[instance]` will starts from 1, so all previous image files will be overwritten. I believe it is better to relate `vm.instance` with `screendump_dir`.